mexbt.go

```diff
@@ -0,0 +1,478 @@
+// meXBT API client
+package mexbt
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	// CreateOrder() sides
+	SIDE_BUY  = "buy"
+	SIDE_SELL = "sell"
+
+	// Product pairs
+	BTCUSD = "BTCUSD"
+	BTCMXN = "BTCMXN"
+
+	// CreateOrder() types
+	ORDER_TYPE_MARKET = 1
+	ORDER_TYPE_LIMIT  = 0
+
+	// Modify() modes
+	MODIFY_BUMP    = 0
+	MODIFY_EXECUTE = 1
+)
+
+var (
+	// Set to true to connect to sandbox API
+	Sandbox = false
+)
+
+type tickerRequest struct {
+	Pair string `json:"productPair"`
+}
+
+type request struct {
+}
+
+type tradesByDateRequest struct {
+	Ins       string `json:"ins"`
+	StartDate int    `json:"startDate"`
+	EndDate   int    `json:"endDate"`
+}
+
+type tradesRequest struct {
+	Ins        string `json:"ins"`
+	StartIndex int    `json:"startIndex"`
+	Count      int    `json:"count"`
+}
+
+type accountTradesRequest struct {
+	StartIndex int `json:"startIndex"`
+	Count      int `json:"count"`
+	signedRequest
+}
+
+type orderbookRequest struct {
+	ProductPair string `json:"productPair"`
+}
+
+type signedRequest struct {
+	ApiKey   string `json:"apiKey"`
+	ApiNonce int64  `json:"apiNonce"`
+	ApiSig   string `json:"apiSig"`
+	Ins      string `json:"ins"`
+}
+
+type createOrderRequest struct {
+	signedRequest
+	Side      string  `json:"side"`
+	OrderType int     `json:"orderType"`
+	Qty       float64 `json:"qty,string"`
+	Px        float64 `json:"px,string"`
+}
+
+type withdrawRequest struct {
+	signedRequest
+	Amount        float64 `json:"amount"`
```

Comment:
Unfortunately the amount has to be a string for this call. This can be only tested in production currently :(